plugin/evm/block.go

```diff
@@ -184,13 +205,18 @@ func (b *Block) VerifyWithContext(ctx context.Context, proposerVMBlockCtx *block
 		log.Debug("Verifying block without context", "block", b.ID(), "height", b.Height())
 	}
 
-	return b.verify(proposerVMBlockCtx, true)
+	return b.verify(&precompileconfig.ProposerPredicateContext{
+		PrecompilePredicateContext: precompileconfig.PrecompilePredicateContext{
+			SnowCtx: b.vm.ctx,
+		},
+		ProposerVMBlockCtx: proposerVMBlockCtx,
+	}, true)
```

Comment:
can we split new predicate context to a variable and then use `b.verify(predicateContext, true)` for readability? we can also create a function for `NewProposerPredicateContext` which would take `ctx` and `proporservmblockctx` as inputs (not really sure how custom we would want this to be though)

Reply:
Going to leave this as is, since I don't think this would really improve readability.